lib/dolly/document.rb

```diff
@@ -74,11 +74,15 @@ def save!
       save
     end
 
-    def destroy soft = false
-      #TODO: Add soft delete support
-      q = id_as_resource + "?rev=#{rev}"
-      response = database.delete(q)
-      JSON::parse response.parsed_response
+    def destroy hard = true
```

Comment:
personally I feel that `document.destroy(true)` should imply a full destroy, whereas `document.destroy(false)` should not.  Thus I changed the name of the arg